### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by maxsokolovsky**
The goal is to ensure the CLI uses the appropriate org for any request.

### CODE REVIEW

File: tests/cli_test.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/shipyard/shipyard-cli/pkg/types"
+	"github.com/shipyard/shipyard-cli/tests/server"
+)
+
+func TestMain(m *testing.M) {
+	cmd := exec.Command("go", "build", "-o", "shipyard", "..")
```
Comment (dietb) [inline tests/cli_test.go:21]: Does this _have_ to build?  That'd leave artifacts lying around.  It seems like it'd make more sense to do these sorts of tests on the functions themselves.

Comment (maxsokolovsky) [reply tests/cli_test.go:21]: Sure, but that'd be not quite end-to-end. Also, TestMain deletes the binary after every run, so there shouldn't be any artifacts lying around.

Comment (roshbhatia) [reply tests/cli_test.go:21]: ++ to what @dietb said here, I'm not sure if we necessarily need to test compilation (that + artifact gen should be done as part of CI anyways, right?), but IMO its not really an issue as it gets cleaned up.

**Review by roshbhatia — APPROVED** [review]
one comment, lgtm otherwise

File: tests/cli_test.go (hunk 1 above)
Comment (maxsokolovsky) [reply tests/cli_test.go:21]: We'd need to keep track of the binary (of a single architecture) in Git, which might be annoying. When there are changes to the code, someone will need to recompile it before running the integration tests. 
It's not about testing compilation but running tests against the **current** code.

Comment (dietb) [reply tests/cli_test.go:21]: We're not getting any benefit from running the compiled binary vs the functions directly from what I can tell.  Meanwhile, the actual functions themselves don't have the test cases attached that were written for this testing.

For the sake of simplicity and avoiding artifact tracking and depending on people to manually, I'd suggest _not_ doing it this way.

Comment (maxsokolovsky) [reply tests/cli_test.go:21]: The main reason for making these tests is ensuring the high-level configuration of the CLI works properly. 
That the config file's values are honored, that env vars properly override those, that the most important flags (such as `org`) are properly seen by our configuration (viper). This used to be a source of bugs. So the tests themselves are made to catch those specific bugs.

I think we could test the functions themselves when we refactor the general setup, such that a look up of anything is just an arbitrary function that can be mocked. But currently, all lookups are `viper.GetString`, which is not easy to test.

What do you think?

Comment (roshbhatia) [reply tests/cli_test.go:21]: Just to expand on what I said earlier -- def shouldn't track the binaries in git, I was referring to tracking them in GitHub itself when creating a release that points to a tag. That being said def out of scope of this, so ignore what I said earlier about it.

**Pull request merged**